src/mbgl/text/bidi.cpp

```diff
@@ -0,0 +1,42 @@
+#include <memory>
+
+#include <mbgl/text/bidi.hpp>
+#include <unicode/ushape.h>
+
+namespace mbgl {
+
+BiDi::BiDi()
+{
+    UErrorCode errorCode = U_ZERO_ERROR;
+    transform = ubiditransform_open(&errorCode); // Only error is failure to allocate memory, in that case ubidi_transform would fall back to creating transform object on the fly
```

Comment:
We're recording the error code, but don't do anything with it. Can we either fail hard here, or add checks in `BiDi::bidiTransform` that don't use the `transform` object if it's not initialized?